lib/Importer/ProtobufLoader.cpp

```diff
@@ -15,18 +15,53 @@
  */
 
 #include "glow/Importer/ProtobufLoader.h"
-
+#include "llvm/Support/CommandLine.h"
 #include <string>
 
 namespace glow {
 
+llvm::cl::OptionCategory loaderOptCat("Model Loader Options");
+
+static llvm::cl::opt<bool> isConstFoldLoaderOps(
+    "const-fold-ops",
+    llvm::cl::desc(
+        "Performs constant folding on ONNX and Caffe Operators while loading."),
+    llvm::cl::init(false), llvm::cl::cat(loaderOptCat));
+
 bool isArrayConstant(llvm::ArrayRef<size_t> a) {
   for (size_t i = 1; i < a.size(); i++)
     if (a[0] != a[i])
       return false;
   return true;
 }
 
+void setConstantFoldLoaderOpsFlag(bool flag) { isConstFoldLoaderOps = flag; }
+
+bool getConstantFoldLoaderOpsFlag() { return isConstFoldLoaderOps; }
+
+bool ProtobufLoader::isConstantFoldable(llvm::SmallVector<NodeValue, 4> &inputs,
```

Comment:
BTW, why does it check only for immediately constant foldable, but would not allow e.g. a recursively constant-foldable patterns like ((c1 + c2) - c3)*c4)

Reply:
The loading happens sequentially. You will always load the input nodes before the current node, so this greedy approach takes care of recursively foldable scenarios.

Reply:
So in this case we would fold `c1 + c2 = c'`, then fold `c' - c3 = c''`, then fold `c'' * c4`. Which means there are multiple compile calls, building multiple temporary Functions/Loaders, etc. It'd be nice to somehow delay the folding to create a single temporary Function/Loader to use like we do with `recursiveClone()` in the graph optimizer, but at first thought this might be a bit complex to handle.

Reply:
I wanted it to be simple, generic enough and less error prone.